File: LeetCode/938. Range Sum of BST.py
###  KNOWLEDGE IS POWER  ###

class Solution:
	def __init__(self):
		self.sum = 0
	def rangeSumBST(self, root: TreeNode, low: int, high: int) -> int:
		if root is None:
			return 
		if root.val >= low and root.val <= high:
			self.sum += root.val

		self.rangeSumBST(root.left, low, high)
		self.rangeSumBST(root.right, low, high)

		return self.sum

File: LeetCode/938. Range Sum of BST.cpp
/***  KNOWLEDGE IS POWER  ***/

class Solution {
public:
    int sumofRange;
    void dfs(TreeNode* root, int L, int R){
		if(!root) 
            return ;
		if(root->val >= L && root->val <= R) 
            sumofRange += root->val;
        if(root->val > L) 
            dfs(root->left,L,R);
        if(root->val < R) 
            dfs(root->right,L,R);
    }
    int rangeSumBST(TreeNode* root, int L, int R) {
        sumofRange = 0;
        dfs(root,L,R);
        return sumofRange;
    }
};
